napari/_qt/containers/_tests/test_qt_layer_list.py

```diff
@@ -32,6 +33,58 @@ def test_set_item_unchecked_makes_layer_invisible(qtbot):
     assert not image.visible
 
 
+def test_alt_click_to_show_single_layer(qtbot):
+    image1 = Image(np.zeros((4, 3)))
+    image2 = Image(np.zeros((4, 3)))
+    image3 = Image(np.zeros((4, 3)))
+
+    layers = LayerList([image1, image2, image3])
+    # this will make the list have image2 on top of image1
+    view = QtLayerList(layers)
+    qtbot.addWidget(view)
+
+    delegate = LayerDelegate()
+
+    # hide the middle-layer, image2 and ensure it's unchecked
+    image2.visible = False
+    assert check_state_at_layer_index(view, 1) == Qt.CheckState.Unchecked
+
+    # ensure the other layers are visible & checked
+    assert image3.visible
+    assert check_state_at_layer_index(view, 0) == Qt.CheckState.Checked
+    assert image1.visible
+    assert check_state_at_layer_index(view, 2) == Qt.CheckState.Checked
+
+    # mock an alt-click on bottom-most layer, image1
+    index = layer_to_model_index(view, 2)
+    delegate._show_on_alt_click_hide_others(view.model(), index)
+
+    # only image1 should be shown, while image3, image2 be hidden
+    assert not image3.visible
+    assert check_state_at_layer_index(view, 0) == Qt.CheckState.Unchecked
+    assert not image2.visible
+    assert check_state_at_layer_index(view, 1) == Qt.CheckState.Unchecked
+    assert image1.visible
+    assert check_state_at_layer_index(view, 2) == Qt.CheckState.Checked
```

Comment:
You mean because the two tests above `test_set_layer_invisible_makes_item_unchecked` and `test_set_item_unchecked_makes_layer_invisible` already ensure there will be no disconnect between states?